pandas/core/tools/datetimes.py

```diff
@@ -156,7 +159,7 @@ def _maybe_cache(arg, format, cache, convert_listlike):
 
 def _box_as_indexlike(
     dt_array: ArrayLike, utc: Optional[bool] = None, name: Optional[str] = None
-) -> Union[ABCIndex, ABCDatetimeIndex]:
+) -> Index:
```

Comment:
@simonjayhawkins do we have a way of annotating a return type is either an Index or DatetimeIndex, but no other Index subclass?

Reply:
not that i'm aware of. If Index subclasses obey Liskov, shouldn't be an issue. why is this needed?

Reply:
> why is this needed?

Not "needed", but i guess its a more-specific-is-better thing